lua/starfall/libs_sv/nextbot.lua

```diff
@@ -0,0 +1,598 @@
+-- @param string? model The model the nextbot will use. If nil, uses Kleiner.
+-- @return NextBot The nextbot.
+function nextbot_library.create(pos, mdl)
+
+	checkpermission(instance, nil, "nextbot.create")
+	checkluatype(mdl, TYPE_STRING)
+	
+	local ply = instance.player
+	nbCount:checkuse(ply, 1)
+
+	local nb = ents.Create("starfall_cnextbot")
+	register(nb, instance)
+	nb:SetPos(vunwrap(pos))
+	nb:SetNBModel(mdl or "models/kleiner.mdl")
+	nb.chip = instance.entity
+	nb:Spawn()
+	table.insert(nextbots, nb)
+	return nbwrap(nb)
+end
+	
+--- Checks if a user can spawn anymore nextbots.
+-- @server
+-- @return boolean True if user can spawn nextbots, False if not.
+function nextbot_library.canSpawn()
+	if not SF.Permissions.hasAccess(instance, nil, "nextbot.create") then return false end
+	return nbCount:check(instance.player) > 0
+end
+
+--- Makes the nextbot try to go to a specified position.
+-- @server
+-- @param Vector gotopos The position the nextbot will continuosly try to go to.
+function nb_methods:setGotoPos(pos)
+		local nb = nbunwrap(self)
+		checkpermission(instance, nb, "nextbot.setGotoPos")
+		nb.goTo = vunwrap(pos)
+end
+
+--- Removes the "go to" position from the NextBot.
+-- @server
+function nb_methods:removeGotoPos()
+	local nb = nbunwrap(self)
+	checkpermission(instance, nb, "nextbot.removeGotoPos")
+	nb.goTo = nil
+end
+
+--- Returns the Vector the nextbot is trying to go to, set by setGotoPos
+-- @server
+-- @return Vector Vector nextbot is trying to go to if it exists, else returns nil.
+function nb_methods:isGoingTo()
+	local nb = nbunwrap(self)
+	if nb.goTo then
+		return vwrap(nb.goTo)
+	else return nil 
+	end
+end
+
+--- Makes the nextbot play a sequence. This takes priority over movement. Will go to set pos after animation plays.
+-- @server
+-- @param string seqtoplay The name of the sequence to play.
+function nb_methods:playSequence(seq)
+	local nb = nbunwrap(self)
+	checkpermission(instance, nb, "nextbot.playSequence")
+	nb.playSeq = seq
+end
+
+--- Makes the nextbot face towards a specified position. Has to be called continuously to be effective.
+-- @server
+-- @param Vector facepos Position to face towards.
+function nb_methods:faceTowards(pos)	
+	local nb = nbunwrap(self)
+	checkpermission(instance, nb, "nextbot.faceTowards")
+	nb.loco:FaceTowards(vunwrap(pos))
+end
+
+--- Sets the activity the nextbot uses for running.
+-- @server
+-- @param number runact The activity the nextbot will use.
+function nb_methods:setRunAct(act)
+	local nb = nbunwrap(self)
+	checkpermission(instance, nb, "nextbot.setRunAct")
+	nb.RUNACT = act
+end
+
+--- Gets the activity the nextbot uses for running.
+-- @server
+-- @return number The run activity.
+function nb_methods:getRunAct(act)
+	local nb = nbunwrap(self)
+	return nb.RUNACT
+end
+
+--- Sets the activity the nextbot uses for idling.
+-- @server
+-- @param number runact The activity the nextbot will use.
+function nb_methods:setIdleAct(act)
+	local nb = nbunwrap(self)
```

Comment:
This and the other functions that take number params need `checkluatype(var, TYPE_NUMBER)`